Formula/luau.rb

```diff
@@ -0,0 +1,21 @@
+class Luau < Formula
+  desc "Fast, safe, gradually typed embeddable scripting language derived from Lua"
+  homepage "https://luau-lang.org"
+  url "https://github.com/Roblox/luau/archive/0.504.tar.gz"
+  sha256 "544b3107e1c53f8349b1e327efed55ac29cc20a2983dd23031ab07c5b6944633"
+  license "MIT"
+  head "https://github.com/Roblox/luau.git", branch: "master"
+
+  depends_on "cmake" => :build
+
+  def install
```

Comment:
```suggestion
  depends_on "cmake" => :build

  on_linux do
    depends_on "gcc"
  end

  fails_with gcc: "5"

  def install
```
This should help with the Linux failure.